Жду разгромного ревью, меня жалеть не надо :)

MaxMaceman opened the pull request:
Единственное, я сильно отстаю по учебе, поэтому будет здорово если критичные замечания будут отдельно от того что я смогу поправить к следующей сдаче ДЗ.

Если я могу как-то оптимизировать сой код, буду рад комментариям.

MaxMaceman committed:
Жду разгромного ревью, меня жалеть не надо :)
Единственное, я сильно отстаю по учебе, поэтому будет здорово если критичные замечания будут отдельно от того что я смогу поправить к следующей сдаче ДЗ.

Если я могу как-то оптимизировать сой код, буду рад комментариям.

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,20 +1,78 @@
+
+
+        if (savedInstanceState != null) {
+            searchQuery = savedInstanceState.getString("search_query", "")
```
Рекомендация - Ключи для передачи данных лучше вынести в константу, т.к. они используются в более чем 1 месте. Ещё один плюс вынесения в константы - в будущем легче будет этот экран переиспользовать

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,20 +1,78 @@
+        outState.putString("search_query", searchQuery)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
```
Рекомендация - `onRestoreInstanceState` можно не переопределять, если ты пользуешься `savedInstanceState` в `onCreate`

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -16,8 +24,67 @@ class SettingsActivity : AppCompatActivity() {
+
+            share_button.setOnClickListener {
+
+                val shareText = "https://practicum.yandex.ru/profile/android-developer/"
```
Ссылку нужно вынести в ресурсы приложения

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -16,8 +24,67 @@ class SettingsActivity : AppCompatActivity() {
+                    type = "text/plain"
+                }
+
+                val chooser = Intent.createChooser(sendIntent, "Поделиться с помощью:")
```
Эту строчку тоже нужно вынести в ресурсы приложения

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -16,8 +24,67 @@ class SettingsActivity : AppCompatActivity() {
+        support_button.setOnClickListener {
+            val email = "maseman.sa@yandex.ru" // Убедитесь, что это правильный адрес
+            val subject = "Сообщение разработчикам и разработчицам приложения Playlist Maker"
+            val body = "Спасибо разработчикам и разработчицам за крутое приложение!"
```
Три этих текста тоже обязательно нужно вынести в ресурсы приложения

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,20 +1,78 @@
+        imm.hideSoftInputFromWindow(inputEditText.windowToken, 0)
+    }
+
+    private fun showKeyboard() {
```
Рекомендация - неиспользуемую функцию лучше удалить

ArturNurtdinov reviewed (commented):
1. Поле ввода на экране поиска не соответствует макетам. Нужно поправить размер иконки лупы, отступ сверху и закругления углов у поля:
![image](https://github.com/user-attachments/assets/8a9aae07-4dd3-4a63-92f0-f969a7a93885)
Макет:
![image](https://github.com/user-attachments/assets/a5e521a2-d29a-4c4f-b7bf-2c59f61f8966)

MaxMaceman committed:
Готово

MaxMaceman committed:
Не правильно скачал векторы из макета...
Поменял иконки - переписал "x_button" и скачал новые "search_page_...", установил отступы согласно макету.
Ещё вынес часть переиспользуемых значений, которых стало больше, в "dimens". :)

MaxMaceman committed:
Не правильно скачал векторы из макета...
Поменял иконки - переписал "x_button" и скачал новые "search_page_...", установил отступы согласно макету.
Ещё вынес часть переиспользуемых значений, которых стало больше, в "dimens". :)

Поспешил, допустил опечатку в "тёмной теме", всё поправил.

ArturNurtdinov reviewed (commented):
Как выглядит экран поиска сейчас:
![image](https://github.com/user-attachments/assets/f11120a3-f088-4bd2-82e8-cfa5a3baaaa2)
![image](https://github.com/user-attachments/assets/bc4b2819-c7fd-41f7-8b0e-e278bd4ec0e7)
Макеты:
![image](https://github.com/user-attachments/assets/166cda27-5149-4ef2-8090-791e5271fc8b)
![image](https://github.com/user-attachments/assets/217d0ec3-df2e-4db6-a5cc-88337a6f51b2)
![image](https://github.com/user-attachments/assets/ce3556b1-71cb-437c-a784-a5f873e7ce41)
![image](https://github.com/user-attachments/assets/274a9077-4462-48ce-94cc-d398f2ad236c)

Для принятия работы необходимо:
1. Поправить UI кнопки "Очистить" в тёмной теме
2. Поправить центрирование заголовка "Поиск"
3. Поправить отступ кнопки "Очистить" от правого края поля ввода - должен быть 12dp
4. Поправить отображение кнопки "Очистить" - кнопка должна отображаться только если поле ввода не пустое

MaxMaceman committed:
Кнопку "Очистить" для себя показывал, пока со стилями работал, забыл скрыть.
Полосу в темной теме убрал, проблема была не в самой кнопке, а в подложке - подложка была серой, а по макету белая, копировал стили из "values" и не заметил что не поменял background.
Заголовок выровнял, кнопку "Очистить" подвинул, сделал отступ от края 8dp.
